framework/decode/vulkan_object_info.h

```diff
@@ -404,6 +405,9 @@ struct SwapchainKHRInfo : public VulkanObjectInfo<VkSwapchainKHR>
     // The following values are only used when loading the initial state for trimmed files.
     std::vector<uint32_t> queue_family_indices{ 0 };
 
+    // For virtual swapchain, we want the compression on virtual images to be the same as on the true swapchain
+    std::optional<VkImageCompressionControlEXT> compression_control;
```

Comment:
Brad has concerns that this isn't forward looking enough and this should be a shared ptr.  I'll look at this and see.  My concern is we can't have a vector of shared pointers because we would have ones for each type.  however, we would only be allocating the size of a pointer for each Swapchain versus an entire structure.

Reply:
Brad did catch something.  There is a pointer to an array in this structure which needs to be re-allocated and freed.  This is a blocker, but I will look at it.

Reply:
Reverted to using a shared_ptr and creating a copy of the flags that are kept around until swapchain destruction.